ex2_hal/adcs/equipment_handler/src/adcs_handler.c

```diff
@@ -545,46 +639,85 @@ ADCS_returnState ADCS_initiate_download_burst(uint8_t msg_length, bool ignore_ho
     command[0] = INITIATE_DOWNLOAD_BURST_ID;
     command[1] = msg_length;
     command[2] = ignore_hole_map;
-    return adcs_telecommand(command, 3);
+    return send_uart_telecommand_no_reply(command, 3);
 }
 
-ADCS_returnState ADCS_receive_download_burst(uint8_t *hole_map, int32_t file_des, uint16_t length_bytes) {
+/**
+ * @brief
+ *      Receives download burst from ADCS and writes packet data to file.
+ * @param hole map
+ *      Tracks missed packets
+ * @param file_des
+ *      File being written to
+ * @param length_bytes
+ *      Length of file being downloaded
+ */
+static ADCS_returnState ADCS_receive_download_burst(uint8_t *hole_map, int32_t file_des, uint16_t length_bytes) {
 
-    if (xSemaphoreTake(adcs_file_download_mutex, UART_TIMEOUT_MS) != pdTRUE) {
-        return ADCS_UART_FAILED;
-    }
+    ADCS_returnState err;
 
-#if defined(USE_UART)
-    for (int i = 0; i < length_bytes / 20; i++) {
-        uint8_t pckt[20] = {0};
-        uint16_t pckt_counter;
-        ADCS_returnState err = receive_file_download_uart_packet(pckt, &pckt_counter);
+    uint16_t num_packets = length_bytes / ADCS_UART_FILE_DOWNLOAD_PKT_DATA_LEN + 1;
+    uint8_t pckt[ADCS_UART_FILE_DOWNLOAD_PKT_DATA_LEN];
+    uint16_t pckt_counter = 0;
+
+    uint8_t dummy_data[20] = {0x33};
+
+    for (int i = 0; i < num_packets; i++) {
+
+        err = receive_file_download_uart_packet(pckt, &pckt_counter);
 
         if (err == ADCS_UART_FAILED) {
-            // End of file
+            // End of file or other error
```

Comment:
Does ADCS not send and end of file marker?

Reply:
Does not